### ISSUE TITLE
Add XGBoost Binary classification inference support (output transformation)

### ISSUE THREAD

**Issue by yanndupis** [tf-encrypted/moose#639]
## Context 

related to #633

For binary classification task, XGBoost model expects to return the predicted class and also the probability associated with each class. See the following example:

```
from xgboost import XGBClassifier
from sklearn.datasets import load_iris

data = load_iris()

X = data.data[((data.target==0) | (data.target==1))]
y = data.target[((data.target==0) | (data.target==1))]

model = XGBClassifier(n_estimators=3, max_depth=2)
model.fit(X, y)

print(model.predict([[6.3, 3.4, 3, 0.5]])) # return [1]
model.predict_proba([[6.3, 3.4, 3, 0.5]]) # return [0.7885864, 0.2114136]
```

## Objectives
- Validate how the aggregation of the outputs of the ensemble of decision trees should be aggregated then transformed to return predicted class and probability associated to each each class. In [this notebook](https://github.com/capeprivacy/research/blob/master/xgboost/xgboost_investigation.ipynb) you will find an example confirming that we need to apply a sigmoid (https://github.com/tf-encrypted/runtime/issues/641) to the output (aggregated decision tree weights) to obtain probability. 
- Extend the EDSL computation developed in the context of regression (#633, #637) task to return predicted class and probability for each class.
